org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/MavenCompletionParticipant.java

```diff
@@ -113,11 +121,46 @@ public void onXMLContent(ICompletionRequest request, ICompletionResponse respons
 					phase -> phase.description, request).forEach(response::addCompletionAttribute);
 			break;
 		case "groupId":
-			if (!parent.getParentElement().getLocalName().equals("parent") && !parent.getParentElement().getLocalName().equals("plugin")) {
-				collectSimpleCompletionItems(localRepositorySearcher.searchGroupIds(RepositorySystem.defaultUserLocalRepository),
+			if (isParentDeclaration) {
+				// TODO local
+				collectParentCompletion(request, response);
+			} else {
+				// TODO if artifactId is set and match existing content, suggest only matching groupId
+				collectSimpleCompletionItems(isPlugin ? localRepositorySearcher.searchPluginGroupIds() : localRepositorySearcher.searchGroupIds(),
 						Function.identity(), Function.identity(), request).forEach(response::addCompletionAttribute);
+				internalCollectRemoteGAVCompletion(request, isPlugin).forEach(response::addCompletionItem);
+			}
+			break;
+		case "artifactId":
+			if (isParentDeclaration) {
+				// TODO local
+				collectParentCompletion(request, response);
+			} else {
+				(isPlugin ? localRepositorySearcher.getLocalPluginArtifacts() : localRepositorySearcher.getLocalArtifactsLastVersion()).keySet().stream()
+					.filter(groupIdArtifactId -> groupId.isEmpty() || groupIdArtifactId.groupId.equals(groupId.get()))
```

Comment:
Should be something like:
`.filter(groupIdArtifactId -> (groupId.isPresent() && groupId.get().isEmpty()) || groupIdArtifactId.groupId.equals(groupId.get()))`

Reply:
same